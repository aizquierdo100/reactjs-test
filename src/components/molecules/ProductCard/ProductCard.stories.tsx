import React, { useState } from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import ProductCard from './ProductCard';

export default {
    title: 'Components/Molecules/ProductCard',
    component: ProductCard,
    argTypes: {
        backgroundColor: { control: 'color' },
    },
} as ComponentMeta<typeof ProductCard>;

const Template: ComponentStory<typeof ProductCard> = (args) => <div className='g--3 g-m--4 g-s--12 fade-in-from-top'><ProductCard {...args} /></div>

export const BaseProductCard = Template.bind({});
BaseProductCard.args = {
    image: 'model2.jpg',
    alt: 'product',
    title: 'Blue mono',
    price: 300000,
    moreInfoHandler: () => {
        alert('Loading...')
    },
    addCartHandler: () => {
        alert('Add to cart')
    }
};

