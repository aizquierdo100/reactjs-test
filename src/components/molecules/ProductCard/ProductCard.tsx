import React, {FC} from "react";

import Button from "../../atoms/Button/Button";
import Card from "../../atoms/Card/Card";
import Label from "../../atoms/Label/Label";
import Image from '../../atoms/Image/Image';
import {renderPrice} from "../../../utils";

interface IProps {
    image: string;
    alt: string;
    title: string;
    price: number;
    moreInfoHandler: () => void,
    addCartHandler: () => void
}

const ProductCard: FC<IProps> = ({image, alt, title, price,
                                     addCartHandler, moreInfoHandler}) => {
    return (
        <Card>
            <React.Fragment>
                <Image src={require(`../../../assets/images/products/${image}`)} alt={alt} />
                <div className='card-content'>
                    <Label className='card-title'>{title}</Label>
                    <Label className='card-subtitle'>{renderPrice(price)}</Label>
                    <div className='container container--between g--12'>
                        <Button onClick={moreInfoHandler}>More info</Button>
                        <Button onClick={addCartHandler}>Add to cart</Button>
                    </div>
                </div>
            </React.Fragment>
        </Card>
    )
}

export default ProductCard;
