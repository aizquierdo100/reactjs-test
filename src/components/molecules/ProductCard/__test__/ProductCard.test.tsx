/**
 *
 * CategoryCard Test
 * @format
 *
 */

import { composeStories } from "@storybook/testing-react";
import * as Stories from "../ProductCard.stories";
import { render, screen } from "@testing-library/react";
import userEvent from '@testing-library/user-event'

const { BaseProductCard } = composeStories(Stories);

describe("ProductCard", () => {
    test("renders BaseProductCard correctly", () => {
        const moreInfoHandler = jest.fn()
        const addCartHandler = jest.fn()
        const view = render(<BaseProductCard moreInfoHandler={moreInfoHandler} addCartHandler={addCartHandler} />);
        expect(view).toMatchSnapshot();
        const addMoreInfoButton = screen.getByText('More info')
        userEvent.click(addMoreInfoButton)
        expect(moreInfoHandler).toHaveBeenCalledTimes(1)
        const addToCart = screen.getByText('Add to cart')
        userEvent.click(addToCart)
        expect(addCartHandler).toHaveBeenCalledTimes(1)
    });
});
