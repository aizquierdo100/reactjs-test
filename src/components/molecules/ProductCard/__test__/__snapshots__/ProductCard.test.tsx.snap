// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard renders BaseProductCard correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="g--3 g-m--4 g-s--12 fade-in-from-top"
      >
        <div
          class="card"
        >
          <img
            alt="product"
            class="img-card"
            src="model2.jpg"
          />
          <div
            class="card-content"
          >
            <p
              class="card-title"
            >
              Blue mono
            </p>
            <p
              class="card-subtitle"
            >
              300,000 VND
            </p>
            <div
              class="container justify--between g--12"
            >
              <button
                class="btn--raised "
              >
                More info
              </button>
              <button
                class="btn--raised "
              >
                Add to cart
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="g--3 g-m--4 g-s--12 fade-in-from-top"
    >
      <div
        class="card"
      >
        <img
          alt="product"
          class="img-card"
          src="model2.jpg"
        />
        <div
          class="card-content"
        >
          <p
            class="card-title"
          >
            Blue mono
          </p>
          <p
            class="card-subtitle"
          >
            300,000 VND
          </p>
          <div
            class="container justify--between g--12"
          >
            <button
              class="btn--raised "
            >
              More info
            </button>
            <button
              class="btn--raised "
            >
              Add to cart
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
